components/match2/wikis/mobilelegends/matchgroup_input_custom.lua

```diff
@@ -0,0 +1,651 @@
+			if counter == 1 and (winner or '') == '' then
+				if finished then
+					winner = scoreIndex
+				end
+			end
+			if lastScore == score then
+				opponents[scoreIndex].placement = tonumber(opponents[scoreIndex].placement or '') or lastPlacement
+			else
+				opponents[scoreIndex].placement = tonumber(opponents[scoreIndex].placement or '') or counter
+				lastPlacement = counter
+				lastScore = score
+			end
+		end
+	end
+
+	return opponents, winner
+end
+
+function CustomMatchGroupInput.placementSortFunction(table, key1, key2)
+	local value1 = tonumber(table[key1].score or _NO_SCORE) or _NO_SCORE
+	local value2 = tonumber(table[key2].score or _NO_SCORE) or _NO_SCORE
+	return value1 > value2
+end
+
+-- Check if any opponent has a none-standard status
+function CustomMatchGroupInput.placementCheckSpecialStatus(table)
+	return Table.any(table, function (_, scoreinfo) return scoreinfo.status ~= 'S' end)
+end
+
+-- function to check for forfeits
+function CustomMatchGroupInput.placementCheckFF(table)
+	return Table.any(table, function (_, scoreinfo) return scoreinfo.status == 'FF' end)
+end
+
+-- function to check for DQ's
+function CustomMatchGroupInput.placementCheckDQ(table)
+	return Table.any(table, function (_, scoreinfo) return scoreinfo.status == 'DQ' end)
+end
+
+-- function to check for W/L
+function CustomMatchGroupInput.placementCheckWL(table)
+	return Table.any(table, function (_, scoreinfo) return scoreinfo.status == 'L' end)
+end
+
+-- Get the winner when resulttype=default
+function CustomMatchGroupInput.getDefaultWinner(table)
+	for index, scoreInfo in pairs(table) do
+		if scoreInfo.status == 'W' then
+			return index
+		end
+	end
+	return -1
+end
+
+--
+-- match related functions
+--
+function matchFunctions.getBestOf(match)
+	match.bestof = Logic.emptyOr(match.bestof, Variables.varDefault('bestof', _DEFAULT_BESTOF))
+	Variables.varDefine('bestof', match.bestof)
+	return match
+end
+
+-- Calculate the match scores based on the map results (counting map wins)
+-- Only update an opponents result if it's
+-- 1) Not manually added
+-- 2) At least one map has a winner
+function matchFunctions.getScoreFromMapWinners(match)
+	local newScores = {}
+	local foundScores = false
+
+	local mapIndex = 1
+	while match['map'..mapIndex] do
+		local winner = tonumber(match['map'..mapIndex].winner)
+		foundScores = true
+		if winner and winner > 0 and winner <= _MAX_NUM_OPPONENTS then
+			newScores[winner] = (newScores[winner] or 0) + 1
+		end
+		mapIndex = mapIndex + 1
+	end
+
+	for index = 1, _MAX_NUM_OPPONENTS do
+		if not match['opponent' .. index].score and foundScores then
+			match['opponent' .. index].score = newScores[index] or 0
+		end
+	end
+
+	return match
+end
+
+function matchFunctions.readDate(matchArgs)
+	if matchArgs.date then
+		local dateProps = MatchGroupInput.readDate(matchArgs.date)
+		dateProps.hasDate = true
+		return dateProps
+	else
+		local suggestedDate = Variables.varDefaultMulti(
+			'matchDate',
+			'Match_date',
```

Comment:
Are these ever set?

Reply:
i think that is a copy paste from another wiki
on sc2 we set one of those in each match (to inherit matchdates)
the other one is iirc the var that old matches set, so useful for during conversion, but not after

if this wiki does not inherit matchDates from match to match then it can be kicked

Reply:
I'll kick this matchdate out and updated this in the 6th update since we likely dont have two matchdates within a single match (have removed this already, thought it would broke |date= but seems to be no effect so 6th update will have these scrapped out)